core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -463,6 +463,27 @@ abstract class SparkFunSuite
       condition = "TABLE_OR_VIEW_ALREADY_EXISTS",
       parameters = Map("relationName" -> tableName))
 
+  protected def roundtripSerialize[T](obj: T): T = {
+    deserializeFromBytes(serializeToBytes(obj))
+  }
+
+  protected def serializeToBytes[T](o: T): Array[Byte] = {
```

Comment:
Do they need to be at the level of `SparkFunSuite`? Shall we just add a base test suite with these methods for the two new suites?

Reply:
sounds good